exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -0,0 +1,249 @@
+  }
+
+  static StackdriverV2ExporterHandler create(String projectId) throws IOException {
+    return new StackdriverV2ExporterHandler(projectId, TraceServiceClient.create());
+  }
+
+  // TODO(bdrutu): Add tests for this.
+  private Span generateSpan(SpanData spanData) {
+    SpanContext context = spanData.getContext();
+    final String traceIdHex = encodeTraceId(context.getTraceId());
+    final String spanIdHex = encodeSpanId(context.getSpanId());
+    SpanName spanName =
+        SpanName.newBuilder().setProject(projectId).setTrace(traceIdHex).setSpan(spanIdHex).build();
+    Span.Builder spanBuilder =
+        Span.newBuilder()
+            .setNameWithSpanName(spanName)
+            .setSpanId(encodeSpanId(context.getSpanId()))
+            .setDisplayName(toTruncatableStringProto(spanData.getName()))
+            .setStartTime(toTimestampProto(spanData.getStartTimestamp()))
+            .setAttributes(toAttributesProto(spanData.getAttributes()))
+            .setTimeEvents(
+                toTimeEventsProto(spanData.getAnnotations(), spanData.getNetworkEvents()))
+            .setStatus(toStatusProto(spanData.getStatus()))
+            .setEndTime(toTimestampProto(spanData.getEndTimestamp()));
+
+    if (spanData.getParentSpanId() != null && spanData.getParentSpanId().isValid()) {
+      spanBuilder.setParentSpanId(encodeSpanId(spanData.getParentSpanId()));
+    }
+
+    return spanBuilder.build();
+  }
+
+  private static String encodeSpanId(SpanId spanId) {
+    return BaseEncoding.base16().lowerCase().encode(spanId.getBytes());
+  }
+
+  private static String encodeTraceId(TraceId traceId) {
+    return BaseEncoding.base16().lowerCase().encode(traceId.getBytes());
+  }
+
+  private static Span.TimeEvents toTimeEventsProto(
+      TimedEvents<Annotation> annotationTimedEvents,
+      TimedEvents<NetworkEvent> networkEventTimedEvents) {
+    Span.TimeEvents.Builder timeEventsBuilder = Span.TimeEvents.newBuilder();
+    timeEventsBuilder.setDroppedAnnotationsCount(annotationTimedEvents.getDroppedEventsCount());
+    for (TimedEvent<Annotation> annotation : annotationTimedEvents.getEvents()) {
+      timeEventsBuilder.addTimeEvent(toTimeAnnotationProto(annotation));
+    }
+    timeEventsBuilder.setDroppedMessageEventsCount(networkEventTimedEvents.getDroppedEventsCount());
+    for (TimedEvent<NetworkEvent> networkEvent : networkEventTimedEvents.getEvents()) {
+      timeEventsBuilder.addTimeEvent(toTimeMessageEventProto(networkEvent));
+    }
+    return timeEventsBuilder.build();
+  }
+
+  private static TimeEvent toTimeAnnotationProto(TimedEvent<Annotation> timedEvent) {
+    TimeEvent.Builder timeEventBuilder =
+        TimeEvent.newBuilder().setTime(toTimestampProto(timedEvent.getTimestamp()));
+    Annotation annotation = timedEvent.getEvent();
+    timeEventBuilder.setAnnotation(
+        TimeEvent.Annotation.newBuilder()
+            .setDescription(toTruncatableStringProto(annotation.getDescription()))
```

Comment:
Truncate to 256 bytes

Reply:
https://github.com/census-instrumentation/opencensus-java/issues/813